work/worker.go

```diff
@@ -495,6 +495,14 @@ func (self *worker) commitNewWork() {
 	defer self.currentMu.Unlock()
 
 	parent := self.chain.CurrentBlock()
+	nextBlockNum := new(big.Int).Add(parent.Number(), common.Big1)
+	var nextBaseFee *big.Int
+	if self.nodetype == common.CONSENSUSNODE {
+		if self.config.IsKIP71ForkEnabled(nextBlockNum) {
+			nextBaseFee = misc.NextBlockBaseFee(parent.Header(), self.config)
+			pending = types.FilterTransactionWithBaseFee(pending, nextBaseFee)
```

Comment:
You mean It's better to merge the two logics?

Reply:
Is there a difference? As far as i understand, those lines are just allocating `pending`. Anyway, is it okay to allocate pending under self.mu & self.currentMu critical section? Because, before hard-fork, the pending is allocated at outside of the section.

Reply:
@yoomee1313 Your question may be related to this discussion (https://github.com/klaytn/klaytn/pull/1442#discussion_r899950415). I thought it's inefficient to traverse the txpool twice. So I suggested to merge two loops (https://github.com/klaytn/klaytn/pull/1442#issuecomment-1158703822). But I also think optimization can be done later.

Reply:
Oh it's not related to performance.

Reply:
@blukat29 thanks for letting me know. Anyway, it looks strange because pending is allocated at different section depending on hard-fork.

Reply:
- It seems the mission of `self.mu` to protect fields under `self`, not local variables.
  Allocation time of `pending` doesn't harm concurrency I guess?
- It's not trivial to move the filtering code above mutex after all. It requires `parent := self.chain.CurrentBlock()` which must be protected by `self.mu`.

Reply:
@blukat29 Thanks for explaining.
@yoomee1313 I think It's rather concerned about txPool mutex, not worker's mutex.
 - txPool.pending() copy all the transactions in pending of txPool and worker use these txs for filtering by BaseFee.
 - the `FilterTransactionWithBaseFee()` has no side effect except its arguments so If we can assure the argument is concurrency safe by worker, I think there's no problem.

Reply:
@blukat29 Oh That's why the pending allocation logic is seperated between critical section. It seems okay to me, however, i'm not sure, too.
@kjeom Thanks for the explanation I understand now.. You mean second allocation is better located in the critical section, right? What about adding a comment about this? It would be helpful to understand why moving is difficult!

Reply:
@yoomee1313 added comments. Please review it again.